### ISSUE TITLE
Common mechanism for building Debug, Debug-NetCore, and Debug-Mono flavors

### ISSUE THREAD

**Issue by cdmihai** [Microsoft/msbuild#508]
On windows and xplat, using the full framework msbuild to build for the `Debug-Mono` fails on Nuget restore.

When #501 gets merged into xplat, building `Debug-Mono` will also fail on Unix due to Nuget restore and MCS issues.

There needs to be a common mechanism for building `Debug`, `Debug-NetCore`, and `Debug-Mono` on Windows and `Debug-NetCore` and `Debug-Mono` on Unix.

**Comment by cdmihai** [comment]
@radical, can you please look into building Debug-Mono on xplat, after #501 gets merged in?

**Comment by radical** [comment]
Sure

**Comment by rainersigwald** [comment]
Closing stale issue--we'll chase down and rediscover Mono build issues as necessary.
